packages/core/src/dispatcher.ts

```diff
@@ -126,13 +126,10 @@ export class Dispatcher {
     // TODO Add validation the message adheres to the proper format.
 
     const { doc: docId, tip } = message
-    const document = await this.repository.get(docId)
-    if (document) {
```

Comment:
It took me a while of digging to figure out why removing this doesn't cause us to now do the work of processing incoming tips for every update message we receive even if the doc isn't in the repository.  I believe the answer is because of how we use the ExecutionQueue within the StateManager and how the ExecutionQueue abandons its task if its `get()` callback returns null, but that's pretty subtle.  I'm honestly not sure the best way to make this more clear, maybe a combination of comments within the ExecutionQueue and the StateManager?

Reply:
That's why ARD and guide are in the pipeline.